tests/Doctrine/ODM/MongoDB/Tests/Tools/EncryptedFieldsMapGeneratorTest.php

```diff
@@ -0,0 +1,261 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Tests\Tools;
+
+use DateTimeImmutable;
+use Doctrine\ODM\MongoDB\Configuration;
+use Doctrine\ODM\MongoDB\DocumentManager;
+use Doctrine\ODM\MongoDB\Mapping\ClassMetadata;
+use Doctrine\ODM\MongoDB\Mapping\ClassMetadataFactoryInterface;
+use Doctrine\ODM\MongoDB\Mapping\MappingException;
+use Doctrine\ODM\MongoDB\MongoDBException;
+use Doctrine\ODM\MongoDB\Tests\BaseTestCase;
+use Doctrine\ODM\MongoDB\Utility\EncryptedFieldsMapGenerator;
+use Doctrine\Persistence\Mapping\AbstractClassMetadataFactory;
+use Doctrine\Persistence\Mapping\Driver\MappingDriver;
+use Doctrine\Persistence\Mapping\ReflectionService;
+use Documents\Bars\Bar;
+use Documents\Encryption\Client;
+use Documents\Encryption\InvalidRootEncrypt;
+use Documents\Encryption\Patient;
+use Documents\Encryption\PatientRecord;
+use Documents\Encryption\RangeTypes;
+use MongoDB\BSON\Decimal128;
+use MongoDB\BSON\UTCDateTime;
+
+use function array_map;
+
+class EncryptedFieldsMapGeneratorTest extends BaseTestCase
+{
+    public function testGetEncryptionFieldsMapForClass(): void
+    {
+        $factory            = new EncryptedFieldsMapGenerator($this->dm->getMetadataFactory());
+        $encryptedFieldsMap = $factory->getEncryptedFieldsMapForClass(Patient::class);
+
+        $expected = [
+            [
+                'path' => 'patientRecord.ssn',
+                'bsonType' => 'string',
+                'keyId' => null,
+                'queries' => ['queryType' => 'equality'],
+            ],
+            [
+                'path' => 'patientRecord.billing',
+                'bsonType' => 'object',
+                'keyId' => null,
+            ],
+            [
+                'path' => 'patientRecord.billingAmount',
+                'bsonType' => 'int',
+                'keyId' => null,
+                'queries' => ['queryType' => 'range', 'min' => 100, 'max' => 2000, 'sparsity' => 1, 'trimFactor' => 4],
+            ],
+        ];
+
+        self::assertEquals(['fields' => $expected], $encryptedFieldsMap);
+    }
+
+    public function testGetEncryptionFieldsMapForClassForEmbeddedDocument(): void
```

Comment:
```suggestion
    public function testGetEncryptionFieldsMapForClassWithEmbeddedDocument(): void
```

I think this name is more descriptive, as I originally expecting this test to query ClientCard directly.